Missing virtual keyword in Advanceable class

```
/home/gromualdi/robot-code/bipedal-locomotion-framework_planner/src/System/include/BipedalLocomotion/System/Advanceable.h:33:14: error: initializer specified for non-virtual method ‘const T& BipedalLocomotion::System::Advanceable<T>::get() const [with T = BipedalLocomotion::Planners::DCMPlannerState]’
     const T& get() const = 0;
              ^~~
/home/gromualdi/robot-code/bipedal-locomotion-framework_planner/src/System/include/BipedalLocomotion/System/Advanceable.h:39:10: error: initializer specified for non-virtual method ‘bool BipedalLocomotion::System::Advanceable<T>::isValid() const [with T = BipedalLocomotion::Planners::DCMPlannerState]’
     bool isValid() const = 0;
          ^~~~~~~
/home/gromualdi/robot-code/bipedal-locomotion-framework_planner/src/System/include/BipedalLocomotion/System/Advanceable.h:45:10: error: initializer specified for non-virtual method ‘bool BipedalLocomotion::System::Advanceable<T>::advance() [with T = BipedalLocomotion::Planners::DCMPlannerState]’
     bool advance() = 0;
          ^~~~~~~
```

--- a/src/System/include/BipedalLocomotion/System/Advanceable.h
+++ b/src/System/include/BipedalLocomotion/System/Advanceable.h
@@ -30,19 +30,19 @@ public:
      * @brief Get the object.
      * @return a const reference of the requested object.
      */
-    const T& get() const = 0;
+    virtual const T& get() const = 0;
 
     /**
      * @brief Determines the validity of the object retrieved with get()
      * @return True if the object is valid, false otherwise.
      */
-    bool isValid() const = 0;
+    virtual bool isValid() const = 0;
 
     /**
      * @brief Advance the internal state. This may change the value retrievable from get().
      * @return True if the advance is successfull.
      */
-    bool advance() = 0;
+    virtual bool advance() = 0;
 };
 
 #endif //BIPEDAL_LOCOMOTION_SYSTEM_ADVANCEABLE_H

